server/src/entity/Transaction.ts

```diff
@@ -0,0 +1,44 @@
+import {
+  BaseEntity,
+  Column,
+  CreateDateColumn,
+  Entity,
+  JoinTable,
+  ManyToMany,
+  ManyToOne,
+  OneToMany,
+  PrimaryGeneratedColumn,
+  PrimaryColumn,
+  OneToOne,
+} from 'typeorm';
+import { Visit } from './Visit';
+
+@Entity()
+export class Transaction extends BaseEntity {
+  @PrimaryGeneratedColumn()
+  public id: string;
+
+  @PrimaryColumn()
+  public visitId: string;
+
+  @PrimaryColumn()
+  public userId: string;
+
+  @OneToMany((type) => Visit, (visit) => visit.transactions)
+  public visit: Visit;
+
+  @PrimaryColumn()
```

Comment:
Why is time a PrimaryColumn? You're using a composite key for 4 values in this entity, seems like id would be the only primary key you need

Reply:
I would also recommend using the `CreateDateColumn` here. http://typeorm.io/#/decorator-reference/createdatecolumn

Reply:
Fixed.